pkg/services/ngalert/state/state.go

```diff
@@ -190,11 +199,11 @@ func resultNormal(state *State, _ *models.AlertRule, result eval.Result, logger
 }
 
 func resultAlerting(state *State, rule *models.AlertRule, result eval.Result, logger log.Logger) {
-	switch state.State {
-	case eval.Alerting:
+	switch {
+	case state.State == eval.Alerting:
 		logger.Debug("Keeping state", "state", state.State)
 		state.Maintain(rule.IntervalSeconds, result.EvaluatedAt)
-	case eval.Pending:
+	case state.State == eval.Pending && state.Error == nil:
```

Comment:
This can make `resultAlerting` a no-op which I don't think is what we want. What's the behaviour we are attempting to create with `state.Error == nil`?

Reply:
This results from having 2 `pending` states, one for `normal` flow and the other for `error`.
We need a way to differentiate both `pending` states and the only thing I came up with without creating other states was to use the error within the state.

This creates a different set of problems where the error pending state is shown in the alert rule 
![Captura de pantalla 2022-12-19 a las 13 33 42](https://user-images.githubusercontent.com/2112640/208601388-4a2ac7ff-17c9-4f9c-8529-2ee39f9224bf.png)
whereas the error state is not.
![Captura de pantalla 2022-12-19 a las 13 35 42](https://user-images.githubusercontent.com/2112640/208601445-7de1198b-1506-459e-aba8-a8d2ebd7a010.png)